lib/ast.js

```diff
@@ -91,7 +91,15 @@ var AST_Token = DEFNODE("Token", "type value line col pos endline endcol endpos
 }, null);
 
 var AST_Node = DEFNODE("Node", "start end", {
-    clone: function() {
+    clone: function(deep) {
+        if (deep) {
+            var self = this.clone();
+            return self.transform(new TreeTransformer(function(node) {
+                if (node !== self) {
+                    return node.clone(true);
```

Comment:
`var self = this.clone();` calls:

https://github.com/alexlamsl/UglifyJS2/blob/0d99a451c77c4786af06c1507648d13776e68121/lib/ast.js#L54-L56

Which is a shallow copy of `props` - simple assignment.